### DISCUSSION TITLE
Is OrangeFS an Object Storage (key-value blob store)?

### DISCUSSION THREAD

**Discussion by pjotrek-b** [waltligon/orangefs#107] [Q&A]
The [German Wikipedia article on OrangeFS](https://de.wikipedia.org/wiki/OrangeFS) says that:

(loosely translated)
> "OrangeFS works object-based: Files in Folders  consist of at least 2 Objects: one for metadata, one for payload."

**Is OrangeFS actually to be considered an Object (Blob?) storage?**

Could it be that it just mimics files-in-folders as keys (similar to S3) - or do I also have an actual Object-Identifier?

And is there any way to access/edit/handle metadata beyond setting filesystem extended attributes?
(It's great! Just wondering if there's more I should now).

thank you very much in advance :smile:

**Comment by waltligon** [comment]
Yes, there is an object-ID. and yes it “mimics” a file system on object blobs.  At the moment there is no way to easily access these OIDs in order to implement a true “blob” based system.  But the code to do that does exist.  We implemented a demo several years back.  It could fairly easily be fleshed out.  The model is a little different - there is an object space underneath each file system entry (complete with owner, permissions, etc.).  So you open the FS entry, authenticate or whatever, and then you have unrestricted access to objects (create, remove, read, write, etc.) under that FS entry.

So the real answer is No, but it is a project away from reality.  Right now Omnibond hasn’t had much demand for such a thing so it isn’t a focus.  Rigth now I’m working on version 3 which has been rearchitected.  When that’s done the object thing could work on that, with replication.

Walt

**Reply by pjotrek-b** [reply]
Thanks for these great insights!
And good to know it'd be theoretically not too-far-away if needed.

I'm currently evaluating which existing "key-value-blob store" implementation to base my own project on.
I came across OrangeFS, because my main focus is on storing real-world metadata with Data Objects - and I honestly didn't know I could setfattr/getfattr to do that.

See my related question: #106

Thanks again, and I'll continue my quest... :smile:
